prow/config/inrepoconfig.go

```diff
@@ -94,23 +95,59 @@ func defaultProwYAMLGetter(
 		return nil, fmt.Errorf("failed to merge: %v", err)
 	}
 
-	prowYAMLFilePath := path.Join(repo.Directory(), inRepoConfigFileName)
-	if _, err := os.Stat(prowYAMLFilePath); err != nil {
-		if os.IsNotExist(err) {
-			log.Debugf("File %q does not exist.", inRepoConfigFileName)
-			return &ProwYAML{}, nil
-		}
-		return nil, fmt.Errorf("failed to check if file %q exists: %v", inRepoConfigFileName, err)
-	}
+	prowYAML := &ProwYAML{}
 
-	bytes, err := ioutil.ReadFile(prowYAMLFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read %q: %v", inRepoConfigFileName, err)
-	}
+	log.Debugf("Attempting to read config files under %q.", inRepoConfigDirName)
+	prowYAMLDirPath := path.Join(repo.Directory(), inRepoConfigDirName)
+	if fileInfo, err := os.Stat(prowYAMLDirPath); !os.IsNotExist(err) && fileInfo.IsDir() {
```

Comment:
`fileInfo` could be nil when other error happens, please catch that before calling `fileInfo.IsDir()`

Reply:
Good catch! Fixed

Reply:
Sorry that I might have not made it super clear.  The case of `err != nil` is still not handled, and we probably don't want to go ahead and read `.prow.yaml` file if `os.Stat` failed with any other reasons. One suggestion here is replacing this line with:
```
if fileInfo, err := os.Stat(prowYAMLDirPath); err == nil && fileInfo.IsDir() {
```

And then after this, do:
```
} else if !os.IsNotExist(err) {
  return fmt.Sprintf("reading %q: %w", prowYAMLDirPath, err)
}
```

Reply:
Sorry, I get it now, already pushed changes PTAL